src/Wpf/Prism.Wpf/Extensions/DependencyObjectExtensions.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+#if HAS_WINUI
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Data;
+#else
+using System.Windows;
+using System.Windows.Data;
+#endif
+
+namespace Prism
+{
+    internal static class DependencyObjectExtensions
+    {
+        public static bool CheckAccess(this DependencyObject instance)
+#if __WASM__
+            // This needs to evolve, once threading is supported
+            // See https://github.com/unoplatform/uno/issues/2302
+            => System.Threading.Thread.CurrentThread.ManagedThreadId == 1;
+#elif HAS_WINUI
+            => instance.Dispatcher.HasThreadAccess;
+#else
+            => instance.CheckAccess();
+#endif
+
+        public static BindingExpression GetBinding(this FrameworkElement instance, DependencyProperty property)
+            => instance.GetBindingExpression(property);
```

Comment:
this seems like all we really do is save having to type out Expression... not really seeing the gain here...

Reply:
Good catch! The original use of this extension is to determine if there is a binding on the property for the object:

https://github.com/PrismLibrary/Prism/blob/3ab9483145e383d85f69a8660835de4a7da80cdf/src/Wpf/Prism.Wpf/Regions/ContentControlRegionAdapter.cs#L38

And GetBinding does not exist in UWP, as the binding needs to get read using `FrameworkElement.GetBindingExpression().ParentBinding`.

That being said, in all cases in Prism, the intent of use is really `HasBinding()`, so all call sites could be replaced by a call to this instead:

```
public static bool HasBinding(this FrameworkElement instance, DependencyProperty property)
#if HAS_WINUI
    => instance.GetBindingExpression(property) != null;
#else
    => BindingOperations.GetBinding(instance, property) != null;
#endif
```

and avoid changing the original call to `BindingOperations.GetBinding`.

Looks good ?